lisp/tree-sitter-debug.el

```diff
@@ -18,9 +18,29 @@
 (defvar-local tree-sitter-debug--tree-buffer nil
   "Buffer used to display the syntax tree of this buffer.")
 
+(defun tree-sitter-debug--button-node-lookup (button)
+  "The function called when `BUTTON' was activated on a tree-sitter debug buffer."
+  (let* ((target-buffer-name (substring (buffer-name) 18)) ; Recover the target buffer name from `tree-sitter-debug--setup'
+         (target-buffer (or (get-buffer target-buffer-name)
+                            (user-error "Could not jump to buffer: \"%s\"" target-buffer-name)))
+         (pos (button-get button 'points-to)))
+    (tree-sitter-debug--goto-node target-buffer (car pos) (cdr pos))))
+
+(defun tree-sitter-debug--goto-node (buffer start end)
+  "Switch to `BUFFER', centering on the region defined by `START' and `END'."
+  (switch-to-buffer-other-window buffer)
+  (goto-char start)
+  (if end
+      (push-mark end t t)
```

Comment:
I think this function shouldn't select the text.

Reply:
Selecting the text is how I figure out where the node ends, since lots of nodes start at the same point. I put it behind a custom option defaulted to false.